### ISSUE TITLE
'No data available' in metrics when filtering with pod-name

### ISSUE THREAD

**Issue by ezieragabriel** [splunk/splunk-connect-for-kubernetes#420]
**What happened**: Splunk Metrics is showing "There is no data available for kube.pod.cpu.usage. To see data on the chart, select a different time range, edit filters, or check with your administrator about user permissions." when adding a "pod-name" filter

**What you expected to happen**: Splunk to show metrics based on filter

**How to reproduce it (as minimally and precisely as possible)**: In Metric workspace, select a metric that can be filtered with pod-name

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-19T22:12:47Z", GoVersion:"go1.12.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-eks-af3caf", GitCommit:"af3caf6136cd355f467083651cc1010a499f59b1", GitTreeState:"clean", BuildDate:"2020-03-27T21:51:36Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}
- Ruby version (use `ruby --version`): ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
- OS (e.g: `cat /etc/os-release`): (Cluster OS) Amazon Linux 2
- Splunk version: 7.3.0
- Others:

**Comment by gp510** [comment]
What version of Splunk Connect for Kubernetes are you using?

**Comment by matthewmodestino** [comment]
can you please review your metric pods and ensure you are not seeing errors? also heres a useful search:

`|mcatalog values(metric_name), values(_dims) WHERE index=foo by host`

**Comment by ezieragabriel** [comment]
@gp510 i'm using splunk connect for k8s v1.4.1 helm charts

**Comment by ezieragabriel** [comment]
> can you please review your metric pods and ensure you are not seeing errors? also heres a useful search:
> 
> `|mcatalog values(metric_name), values(_dims) WHERE index=foo by host`
> 
> Where `host` is the collector/node name

@matthewmodestino no errors in metric pods. The search string also returned the expected result, I believe.

**Comment by ezieragabriel** [comment]
I discovered that the graph error is caused by the ' (quote) added in the metric name when I tried to open the graph in search.

This is the metric i'm trying to view in metric workspace. It works until i add a pod-name filter.
![image](https://user-images.githubusercontent.com/11989821/87230269-576c6180-c3e1-11ea-8194-1ad25cc37ba5.png)

I tried to open it in search and this is the generated search string
![image](https://user-images.githubusercontent.com/11989821/87230281-72d76c80-c3e1-11ea-857f-a0d1d7180efd.png)
notice the quotes around the pod-name

When the quotes are removed, then the chart appears successfully
![image](https://user-images.githubusercontent.com/11989821/87230306-b16d2700-c3e1-11ea-8ff7-25d16031cb71.png)

(Some values are omitted for security)
